generator/src/test/java/com/scottlogic/deg/generator/inputs/ProfileReaderTests.java

```diff
@@ -534,4 +534,18 @@ public void shouldRejectIsConstraintSetToNull() {
 
         expectInvalidProfileException();
     }
+
+    @Test
+    public void shouldRejectIsConstraintSetToNullForNot() {
+        givenJson(
+            "{" +
+                "    \"schemaVersion\": \"v3\"," +
+                "    \"fields\": [ { \"name\": \"foo\" } ]," +
+                "    \"rules\": [" +
+                "        { \"not\": { \"field\": \"foo\", \"is\": null } }" +
+                "    ]" +
+                "}");
+
```

Comment:
Hi, perhaps we should add a test for both possible versions of rule deserialisation - see RuleDeserializer in schemas project.